### PR TITLE
Read all-electron WFN and uncontracted basis from CP2K

### LINKED ISSUE #41: Improved IO code for CP2K ATOM output
I have an improved reader for CP2K calculations. It supports AE and PP densities, and contracted and uncontracted basis sets.

### PR DESCRIPTION
**Pull request by tovrstra**
Fixes #41 

With this code, all different types of CP2K ATOM calculations can be read into HORTON.

### CODE REVIEW

**Comment by tovrstra** [comment]
Ready for review.

File: horton/test/common.py [hunk 1]
```diff
@@ -372,3 +376,32 @@ def numpy_seed(seed=1):
...
+              The file to be truncated.
+    nline : int
+            The number of lines to retain.
+    nadd : int
+           The number of empty lines to add.
+    """
+    with tmpdir(name) as dn:
+        fn_truncated = '%s/truncated_%i_%s' % (dn, nline, os.path.basename(fn_orig))
+        with open(fn_orig) as f_orig, open(fn_truncated, 'w') as f_truncated:
+            counter = 0
```
Comment (matt-chan) [inline horton/test/common.py]: I think this can be more pythonic if you use an enumerate instead of the explicit counter?

**Comment by tovrstra** [comment]
I will try to fix the merge conflicts by merging the branch manually (instead of rebasing). That should also work...

**Comment by tovrstra** [comment]
That worked and it is probably nicer than rebasing.

**Comment by matt-chan** [comment]
Looks good. Merging!

**Pull request merged**